packages/graphback-runtime-mongodb/src/OffixMongoDBDataProvider.ts

```diff
@@ -10,17 +10,50 @@ import { MongoDBDataProvider } from './MongoDBDataProvider';
  * https://offix.dev/docs/conflict-server#structure-of-the-conflict-error
  */
 export class OffixMongoDBDataProvider<Type = any, GraphbackContext = any> extends MongoDBDataProvider<Type, GraphbackContext> {
+  protected updatedAtField: string;
+  protected createdAtField: string;
 
   public constructor(baseType: GraphQLObjectType, client: any) {
     super(baseType, client);
+    this.createdAtField = undefined;
+    this.updatedAtField = undefined;
+    Object.keys(baseType.getFields()).forEach((k: string) => {
+      if (baseType.getFields()[k]?.astNode?.directives) {
+        baseType.getFields()[k].astNode.directives.forEach((directive: any) => {
+          if (directive?.name?.value === "createdAt") {
+            if (this.createdAtField === undefined) {
+              this.createdAtField = baseType.getFields()[k]?.name;
+            } else {
+              throw Error("Cannot have more than one field with createdAt directive");
+            }
+          }
+          if (directive?.name?.value === "updatedAt") {
+            if (this.updatedAtField === undefined) {
+              this.updatedAtField = baseType.getFields()[k]?.name;
+            } else {
+              throw Error("Cannot have more than one field with updatedAt directive");
+            }
+          }
+        });
+      }
+    });
   }
 
   public async create(data: any): Promise<Type> {
     if (!data.version) {
       data.version = 1;
     }
+    const o = await super.create(data);
+    let res = o as any;
+    if (this.updatedAtField) {
+      res = await super.update({
+        ...res,
+        id: res.id,
+        [this.updatedAtField]: (new ObjectId(res.id)).getTimestamp()
+      });
+    }
 
-    return super.create(data);
+    return this.mapFields(res);
   }
```

Comment:
Can we issue one query to mongo here? 

I am thinking something like 
```javascript
const creationDate = new Date(); // or Date.now()
return super.create({
...data,
[this.createdAtField]: this.createdAtField ? creationDate : undefined,
[this.updatedAtField]: this.updatedAtField ? creationDate : undefined
}) 
```

Reply:
Currently, there is no actual createdAt field in the actual documents, it is a pseudo-field?(help me) 
I am getting creation timestamp from the ObjectId of the document see [this](https://docs.mongodb.com/manual/reference/method/ObjectId/#ObjectId)
https://github.com/aerogear/graphback/blob/d50fcdbec9b2d6e7b36250c13c301a156b3d5666/packages/graphback-runtime-mongodb/src/OffixMongoDBDataProvider.ts#L91
So at this point we could very well store both fields, I am not against that approach if it aligns more with what we are doing/ :)

Reply:
Out of curiosity what and why is this bad idea? Using ObjectId as creation timestamp is the only way to ensure that this time is accurate. I will check what mongose folks are doing but I guess they would do something similar

Reply:
I was wrong - they use `new Date()`
https://mongoosejs.com/docs/guide.html#timestamps

Reply:
I think if somebody uses graphback as their primary crud tool, they will be surprised if they didn't find any created at field in the actual document. 

Also if we go with creating an actual field, there is more flexibility: if somebody doesn't want to use extra field they could implement this objectID-to-timestamp approach  (in client side maybe) and just omit the @createdAt directive from schema (only possible for mongo and not postgres). If someone is working with mongo extensively, they would normally know this and use this as well.

We also have option of creating new field using the createdAt directive (I am mulling over what advantages this has besides simplifying the the [filtering code](https://kchodorow.com/2011/12/20/querying-for-timestamps-using-objectids/) a **lot** and perhaps making the data more portable? I'm not sure), but I don't think this has any significant disadvantages at least (besides using more space)

I would also assume the new field option would be less prone to bugs (since less code actually involved), plus easier to test as well.

I am kinda impartial to both so would love some opinions!

Reply:
Ok. We have some prototype and investigation. Let's step back to the issue and explain how this might work (for both DBs)